### ISSUE TITLE
400 bad request

### ISSUE THREAD

**Issue by fdddf** [cliff363825/yii2-kindeditor#1]
KindEditorBaseController

``` php
if (Yii::$app->request->enableCsrfCookie) {
            Yii::$app->response->cookies->add(new \yii\web\Cookie([
                'name'=>Yii::$app->request->csrfParam,
                'value'=>Yii::$app->request->post($this->csrfCookieParam)
            ]));
//             $_COOKIE[Yii::$app->request->csrfParam] = Yii::$app->request->post($this->csrfCookieParam);
}
```

Upload an image using this lib, it reports http 400 error. I fixed the issue with above code. But it was not fixed for multiple image uploading. It has the same issue.
So I have to set enableCsrfValidation as false, it works fine.
Could you please find why? Thx.

**Comment by cliff363825** [comment]
thank you for reporting it . 
i have solved it.

KindEditorBaseController
code:

``` php
    /**
     * @inheritdoc
     */
    public function init()
    {
        parent::init();
        if (($sessionId = Yii::$app->request->post(Yii::$app->session->name)) !== null) {
            Yii::$app->session->setId($sessionId);
            Yii::$app->session->open();
        }
        if (Yii::$app->request->enableCsrfCookie) {
            // fix bug #1: 400 bad request [by fdddf]
            if (!isset($_COOKIE[Yii::$app->request->csrfParam])) {
                $_COOKIE[Yii::$app->request->csrfParam] = Yii::$app->request->post($this->csrfCookieParam);
            }
        }
    }
```

**Comment by fdddf** [comment]
@cliff363825 Great job! Thx.
